File: src/main/java/service/task/management/service/TaskServiceImplementation.java
package service.task.management.service;

import service.task.management.model.Task;
import service.task.management.repository.TaskRepository;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class TaskServiceImplementation implements TaskService {

    @Autowired
    private TaskRepository repository;

    @Override
    public List<Task> findAll() {

        List<Task> tasks = (List<Task>)repository.findAll();

        return tasks;
    }
}

File: src/main/kotlin/service/task/management/service/TaskServiceImplementation.kt
package service.task.management.service

import com.google.inject.Inject
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import service.task.management.model.Task
import service.task.management.repository.TaskRepository

@Service
class TaskServiceImplementation @Inject constructor(
    val taskRepository: TaskRepository
) : TaskService {
    override fun findAll(): List<Task> {
        return taskRepository.findAll().toList()
    }

    override fun createTask(task: Task): Task {
        return taskRepository.save(task)
    }
}